x-pack/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -220,379 +198,410 @@ export function App({
       }
     });
   }, [
-    lastKnownDoc,
     onAppLeave,
-    state.persistedDoc,
+    lastKnownDoc,
     state.isSaveable,
-    core.application.capabilities.visualize.save,
+    state.persistedDoc,
     getLastKnownDocWithoutPinnedFilters,
+    application.capabilities.visualize.save,
   ]);
 
   // Sync Kibana breadcrumbs any time the saved document's title changes
   useEffect(() => {
-    core.chrome.setBreadcrumbs([
-      ...(originatingApp && getAppNameFromId
-        ? [
-            {
-              onClick: (e) => {
-                core.application.navigateToApp(originatingApp);
-              },
-              text: getAppNameFromId(originatingApp),
-            } as EuiBreadcrumb,
-          ]
-        : []),
-      {
-        href: core.http.basePath.prepend(`/app/visualize#/`),
+    const byValueMode =
+      // Temporarily required until the 'by value' paradigm is default.
+      dashboardFeatureFlag.allowByValueEmbeddables &&
+      state.isLinkedToOriginatingApp &&
+      !(initialInput as LensByReferenceInput)?.savedObjectId;
+    const breadcrumbs: EuiBreadcrumb[] = [];
+    if (state.isLinkedToOriginatingApp && getOriginatingAppName() && redirectToOrigin) {
+      breadcrumbs.push({
+        onClick: () => {
+          redirectToOrigin();
+        },
+        text: getOriginatingAppName(),
+      });
+    }
+    if (!byValueMode) {
+      breadcrumbs.push({
+        href: application.getUrlForApp('visualize'),
         onClick: (e) => {
-          core.application.navigateToApp('visualize', { path: '/' });
+          application.navigateToApp('visualize', { path: '/' });
           e.preventDefault();
         },
         text: i18n.translate('xpack.lens.breadcrumbsTitle', {
           defaultMessage: 'Visualize',
         }),
-      },
-      {
-        text: state.persistedDoc
-          ? state.persistedDoc.title
-          : i18n.translate('xpack.lens.breadcrumbsCreate', { defaultMessage: 'Create' }),
-      },
-    ]);
+      });
+    }
+    let currentDocTitle = i18n.translate('xpack.lens.breadcrumbsCreate', {
+      defaultMessage: 'Create',
+    });
+    if (state.persistedDoc) {
+      currentDocTitle = byValueMode
+        ? i18n.translate('xpack.lens.breadcrumbsByValue', { defaultMessage: 'Edit Visualization' })
```

Comment:
Sentence case should be used here:

```suggestion
        ? i18n.translate('xpack.lens.breadcrumbsByValue', { defaultMessage: 'Edit visualization' })
```

Unless you wanted to make this say something like "Edit Lens"

Reply:
Since the plan is to have this breadcrumb appear consistent between visualize and lens, I think the sentence case 'Edit visualization' works quite well, because even there it will be a noun rather than an app title. Good call